### ISSUE TITLE
Bootstrap 4 based version of WET-BOEW

### ISSUE THREAD

**Issue by michael-milette** [wet-boew/wet-boew#7804]
Any idea when a Bootstrap 4 based version of WET-BOEW might become available? Last I heard was that something was in the works back in 2015.

If we are still far away from this, does anyone have any experience with attempting to just dropping Bootstrap 4 in as a replacement for Bootstrap 3 in this framework?

**Comment by kodecount** [comment]
To start WET with it current setup could only go from [Bootstrap 3.3.1](https://github.com/wet-boew/wet-boew/blob/master/bower.json#L25) to [Bootstrap 3.3.7](https://github.com/twbs/bootstrap-sass) since that is the latest version of Bootstrap Sass. In order to move to Bootstrap 4 WET would require to change it bootstrap library to Ruby gem.

Also the other thing testing for styling, accessibility, or functionality issues brought by the change, since their is a large amount of difference just in 3.3.1 to 3.3.7.

**Comment by kodecount** [comment]
With the current setup WET could only go from [Bootstrap 3.3.1](https://github.com/wet-boew/wet-boew/blob/master/bower.json#L25) to [Bootstrap 3.3.7](https://github.com/twbs/bootstrap-sass) since that is the latest version of Bootstrap Sass. In order to move to Bootstrap 4 WET would require to change it bootstrap library to Ruby gem (this be a major overhaul I believe).

Also the other thing is testing for styling, accessibility, or functionality issues brought by the change, since their is a large amount of difference just in 3.3.1 to 3.3.7.

If you want to update to 3.3.7 you can just change Line 25 in the `bower.json` I believe.

I can't see WET moving to Bootstrap 4 anytime soon. Also considering it still in alpha and completely changes the backbone of WET.

**Comment by duboisp** [comment]
@michael-milette as far as I am aware, I can't tell you when Bootstrap 4 will be available in wet-boew. Considering there will be a lot of testing and a lot of adaptation to do, I would assume the probability is very high that would only happen in the next full major revision of WET. 

I know that Laurent and Mario did a demo at Codefest 4 in their session "the future of WET" where they show new plugin structure, etc., but today we are still in exploration/experimentation phase. I am not aware of any official migration plan yet.

Cheers

**Comment by LaurentGoderre** [comment]
Yep! No major upgrade of Bootstrap without a major upgrade to WET itself.

**Comment by michael-milette** [comment]
What are or where can I learn more about the current plans for the next version of WET?

**Comment by LaurentGoderre** [comment]
That I don't know. I created a prototype I shown at last codefest but I haven't heard anything since.

**Comment by duboisp** [comment]
@michael-milette I don't know.

**Comment by nschonni** [comment]
It's still in an Alpha state, but this gives a good comparison about what's changed https://medium.com/wdstack/bootstrap-4-whats-new-visual-guide-c84dd81d8387#.uk7it4s4i
I think the biggest problem will be the changes to the grid system since the breakpoints have change (probably for the better) which would require people to redo alot of their layouts.

**Comment by michael-milette** [comment]
My challenge is that I need to integrate the WET theme with an existing open source application which only has base themes available for bootstrap 2 and bootstrap 4. I am running into some conflicts running two different versions of bootstrap simultaneously. I suspect that, as time goes on, people will be migrating to bootstrap 4. Developing for a soon to be deprecated framework isn't a long term sustainable solution.

There were a lot of big advantages presented for the next version of WET which were described to the audience at the last WET CodeFest 4 back in February of this year. Shortly after that, CodeSprints seem to have stopped as as well as any innovation of the WET framework.

I see on the [roadmap](http://wet-boew.github.io/v4.0-ci/docs/versions/rdmp-en.html) for the WET-BOEW framework that a release of WET 4.1.0 is expected in May 2017. How can I learn more about this release?

**Comment by nschonni** [comment]
It's odd that anything is requiring 4 at all right now since there are breaking changes between releases since it is still in the alpha stage.

**Comment by kodecount** [comment]
I agree it seems way to early to be pushing Bootstrap 4 to production when it is in alpha and bugs are still being worked out.

**Comment by kodecount** [comment]
@michael-milette What is this "open source application" ?

**Comment by nschonni** [comment]
OK, they've hit Beta now so I'm :100: on starting to look at this now http://blog.getbootstrap.com/2017/08/10/bootstrap-4-beta/
This would be a big breaking change, so it would probably need to be a WET 5 thing

**Comment by Bgolden67** [comment]
@nschonni .. wet5 - ouch! massive conversion projects ahead.
I **won't** quote you on this but do you have a rough target in mind as to when you THINK we MIGHT possibly see WET5?

**Comment by nschonni** [comment]
@Bgolden67 I have no idea if/when WET 5 would happen. Just with the scope of very core changes in Bootstrap 4, I don't think it would work as a side-by-side addition in a WET 4.x release.

**Comment by michael-milette** [comment]
I've heard rumours of a WET 5 potentially coming this year. Any idea what HTML/JavaScript/CSS framework will be used this time?

**Comment by duboisp** [comment]
@michael-milette we are still in a early stage of thinking about WET5, there is nothing concrete at this moment.

**Comment by michael-milette** [comment]
Thanks @duboisp , do you know if it will likely be based on Bootstrap 4 or some other HTML / CSS / JavaScript framework?

**Comment by olstjos** [comment]
https://github.com/wet-boew/wet-boew/issues/9508
